packages/next/src/build/entries.ts

```diff
@@ -119,17 +117,21 @@ export async function getStaticInfoIncludingLayouts({
     pageFilePath,
     isDev,
     page,
+    // TODO: sync types for pages: PAGE_TYPES, ROUTE_TYPES, 'app' | 'pages', etc.
     pageType: isInsideAppDir ? PAGE_TYPES.APP : PAGE_TYPES.PAGES,
   })
 
-  const staticInfo: PageStaticInfo = isInsideAppDir
-    ? {
-        // TODO-APP: Remove the rsc key altogether. It's no longer required.
-        rsc: 'server',
-      }
-    : pageStaticInfo
+  if (!isInsideAppDir || !appDir) {
+    return pageStaticInfo
+  }
 
-  if (isInsideAppDir && appDir) {
+  const staticInfo: PageStaticInfo = {
+    // TODO-APP: Remove the rsc key altogether. It's no longer required.
+    rsc: 'server',
+  }
+
+  // inherit from layout files only if it's a page file
+  if (isAppPageFile(pageFilePath, pageExtensions)) {
```

Comment:
You can just check `page` to see if it's ends with `/page`, using existing helper `isAppRoute`

Reply:
Then we should pass the `originalAppPath`, so as other entries that use this API:

```
{
  page: '/app-layout/inner',
  pageFilePath: '/private/var/folders/cm/.../app/app-layout/inner/page.tsx',
  originalAppPath: '/app-layout/inner/page'
}
```

<img width="847" alt="Screenshot 2024-08-14 at 6 32 06 PM" src="https://github.com/user-attachments/assets/986e7f42-7924-4272-93f8-9ddebefc04b6">

Reply:
https://github.com/vercel/next.js/pull/68793/files#diff-90d1d5f446bdf243be25cc4ea2295a9c91508859d655e51d5ec4a3562d3a24d9R2032

Passed `originalAppPath` if exists or else `page`.